app/common/urlParse.js

```diff
@@ -14,7 +14,23 @@ module.exports = (url, ...args) => {
     return Object.assign({}, parsedUrl)
   }
 
-  parsedUrl = urlParse(url, ...args)
+  // In fast-url-parser href, port, prependSlash, protocol, and query
+  // are lazy so access them.
+  const raw = urlParse(url, ...args)
+  parsedUrl = {
```

Comment:
The first 5 properties are lazy / accessor functions, so this is to populate those properties.